src/hooks/useCedarContactLookup.tsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useApolloClient } from '@apollo/client';
+
+import GetCedarContactsQuery from 'queries/GetCedarContactsQuery';
+import { CedarContactProps } from 'types/systemIntake';
+
+function useCedarContactLookup(
+  query: string
+): { [id: string]: CedarContactProps };
+
+// eslint-disable-next-line no-redeclare
+function useCedarContactLookup(
+  query: string,
+  euaUserId: string
+): CedarContactProps | undefined;
+
+// eslint-disable-next-line no-redeclare
+function useCedarContactLookup(
+  query: string,
+  euaUserId?: string
+): { [id: string]: CedarContactProps } | CedarContactProps | undefined {
+  const client = useApolloClient();
+  const [cedarContacts, setCedarContacts] = useState<CedarContactProps[]>([]);
+  const contactByEuaUserId = useMemo(() => {
+    return cedarContacts.find(contact => contact.euaUserId === euaUserId);
+  }, [euaUserId, cedarContacts]);
+
+  useEffect(() => {
```

Comment:
For the frontend folks who're more familiar with Apollo/React internals - is there any issue integrating this custom hook that calls `useEffect()` with however the Apollo client works?

Reply:
As far as I know this wasn't a problem necessarily, but I did end up refactoring it anyway to not use `useEffect`. The original code was a modified version of a hook used in Mint.